Lib/test/test_asyncio/test_subprocess.py

```diff
@@ -182,6 +183,29 @@ def test_kill(self):
         else:
             self.assertEqual(-signal.SIGKILL, returncode)
 
+    def test_kill_issue43884(self):
+        import subprocess
+        blocking_shell_command = f'{sys.executable} -c "import time; time.sleep(100000000)"'
+        creationflags = 0
+        if sys.platform == 'win32':
+            # On windows create a new process group so that killing process
+            # kills the process and all its children.
+            creationflags = subprocess.CREATE_NEW_PROCESS_GROUP
+        proc = self.loop.run_until_complete(
+            asyncio.create_subprocess_shell(blocking_shell_command, stdout=asyncio.subprocess.PIPE,
+            creationflags=creationflags)
+        )
+        self.loop.run_until_complete(asyncio.sleep(1))
+        if sys.platform == 'win32':
+            proc.send_signal(signal.CTRL_BREAK_EVENT)
+        proc.kill()
+        returncode = self.loop.run_until_complete(proc.wait())
+        if sys.platform == 'win32':
+            self.assertIsInstance(returncode, int)
+            # expect 1 but sometimes get 0
```

Comment:
This is a bit disappointing. Do you know what makes it return one or the other?

Reply:
I have just copy pasted this from other tests, also subprocess killing on Windows is racy at the kernel level (No proof but this is what I have heard.)